nml/expression/spritegroup_ref.py

```diff
@@ -72,6 +73,13 @@ def get_action2_id(self, feature):
         return spritegroup.get_action2(feature).id
 
     def reduce(self, id_dicts = [], unknown_id_fatal = True):
+        if self.name.value != 'CB_FAILED':
+            spritegroup = action2.resolve_spritegroup(self.name)
+            if isinstance(spritegroup, switch.Switch) and spritegroup.expr.is_read_only() and \
+                    spritegroup.body.default is not None and spritegroup.body.default.value is not None and \
+                    len(spritegroup.body.ranges) == 0:
+                generic.print_warning("Block '{}' returns a constant, optimising.".format(spritegroup.name.value), self.pos)
```

Comment:
don't know how nml prints work - i would expect this to be a debug message, rather than a warning?

Reply:
A switch with no conditions (returning only its default value) is rather pointless, so I think the idea is to warn the grf author in case it's a mistake and they forgot something.

I'm not convinced that's correct - the "generated by dumb script" explanation does exist.

Reply:
If this turns out to be spammy and hard to solve due to dumb scripts, should we introduce a way to disable the warning?

Reply:
0.4 compatibility already added many warnings. And there's already a way to disable the warnings (`--quiet`), but it's not selective :). I think we need a way to filter warnings.

Reply:
Adding a way to filter warnings sounds like a good plan, but does not belong in this PR.